Fix: Build failure on Node.js v22 and add an updateUrl interface for dynamic URL updates, such as time changes.

Oseenix opened the pull request.

Oseenix committed:
add updateUrl interface, eg. url change as time change

Oseenix committed:
fix: build fail in node.js v22, update canvas

Oseenix committed:
fix package-lock.json conflict

Oseenix committed:
pkgs update

eddy-geek commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "maplibre-contour",
-  "version": "0.1.0",
+  "name": "@oseenix/maplibre-contour",
```
this shouldn't be in the PR ?

Oseenix replied:
Yes, you are right, the pr was not accepted, so i published in github pkgs, and forgot this PR

hongruping committed:
fix:调整 maplibre-contour 插件，在 DEM URL 更新后刷新 contour protocol URL，避免MapLibre 复用旧等高线矢量瓦片。
fix:修改版本为0.1.2

hongruping committed:
fix:优化 maplibre-contour 等高线 source 切换：新增setSource(url, { key }) 和请
    求级 source snapshot，避免快速切换时新旧 raster URL 串扰。
fix:调整 contour 缓存键为 sourceKey + z/x/y + options，保证不同 forecast step 缓
    存隔离，同一 step 可复用。
